scipy/interpolate/_rgi_cython.pyx

```diff
@@ -56,10 +149,14 @@ def find_indices(grid, bounds_error, xi):
 
     # iterate through dimensions
     for i in range(xi.shape[0]):
-        index = np.searchsorted(grid[i], xi[i]) - 1
-        index[index < 0] = 0
-        index[index > grid[i].size - 2] = grid[i].size - 2
-        indices[i] = index
+        grid_i = grid[i]
+        # index = np.searchsorted(grid[i], xi[i]) - 1
+        # index[index < 0] = 0
+        # index[index > grid[i].size - 2] = grid[i].size - 2
+        # indices[i] = index
+        for j in range(xi.shape[1]):
+            index = find_interval_descending(&grid_i[0], xi.shape[1], xi[i][j]) - 1
+            indices[i][j] = index
```

Comment:
Here I've omitted the extra logic from the commented out lines. This still needs to be implemented

Reply:
This is now done